real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/localizers/HighTrustGPSLocalizer.java

```diff
@@ -35,27 +37,37 @@ public HighTrustGPSLocalizer(){
         buggyFrameGpsLon = 0.0;
         buggyFrameGpsLat = 0.0;
         posePub = new Publisher(NodeChannel.POSE.getMsgPath());
-
+        mostRecentUpdate = new Date();
 
 
         //Initialize subscriber to GPS measurements
         new Subscriber(NodeChannel.GPS.getMsgPath(), new MessageListener() {
             @Override
             public void actionPerformed(String topicName, Message m) {
-                GpsMeasurement newGPSData = (GpsMeasurement)m;
-
+            	GpsMeasurement newGPSData = (GpsMeasurement)m;
+            	synchronized (this) {
+            	long dt = newGPSData.getTimestamp().getTime() - mostRecentUpdate.getTime();
+            	System.out.println(dt);
+            	if(dt > 0.0){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2325208361/issues/source?bid=2938665&fileBranchId=3199623#l51)